### PR TITLE
Change root directory from monorepo to Vue CLI project

### LINKED ISSUE #905: [RFC] rearchitect project to keep separate packages with single Storybook app
## Problem

Our project is currently split into multiple packages each with its own Storybook instance. The separate packages are desirable so we can use them in downstream projects. However, having multiple Storybook instances has led to some breaking complexity and redundant configuration. 

In effect, the complexity introduced by the underlying dependencies has become an impediment to further development.

Additionally, the package documentation should probably not be intermixed with package code (unless in the form of comments/docstrings).

## Description

Determine how to rearchitect this project so that we have

- only one global Storybook project and
- separate packages for vocabulary, fonts, and vue-components (allowing packaged distribution)

With this architecture, we might then be able to deprecate `lerna`.

## Example architecture

- `/src/` or `/packages/` - contains the sub-packages
    -  `fonts/` - fonts package (can possibly be merged into the `vocabulary` package)
    - `vocabulary/` - vocabulary package (possibly including the `fonts` package)
    - `vue-vocabulary/` - Vue components
- `/docs/` - documentation project containing Storybook code
    - `package.json` - dependencies for Storybook project
    - imports for vocabulary packages might be defined in package.json or `index.js`

## Related issue(s)
- https://github.com/creativecommons/vocabulary/issues/630

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #905 by @brylie 

## Description
This PR rearchitects the project to make it more maintainable.

## Todo

Fix remaining components:

- [x] Donate Button
- [x] VButton
- [x] VCheckbox
- [x] VHeader
- [x] VRadio
- [x] Tabs
- [x] VLocale
- [x] VModal

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
The Storybook is up and the story about the SVG icon font has been completely rewritten to use Storybook's built in icon gallery. It looks fantastic!

**Comment by brylie** [comment]
This looks good so far. Quite a lot of cleanup!

**Comment by dhruvkb** [comment]
@brylie due to time constraints, I'll only be able to work further on this during the weekend. In the meantime feel free to push commits to the branch 👍!

**Comment by brylie** [comment]
OK, thanks @dhruvkb. No hurry :smiley:

**Comment by dhruvkb** [comment]
Assimilated Vocabulary stories into Storybook. A lot of things are broken because of broken paths but they should be fixed pretty soon. It's cool that most stories are visible now.

**Comment by dhruvkb** [comment]
I've fixed two component stories in 56e2706. @brylie if you have some time, could you replicate this across the other broken components?

**Comment by brylie** [comment]
@sadn1ck has expressed interest in helping us get this pull request to the finish line :checkered_flag: :1st_place_medal:  

@dhruvkb would you kindly help @sadn1ck with any code-related difficulties they might face? :man_technologist: 

As was mentioned, [commit `56e2706` serves as a recipe to follow](https://github.com/creativecommons/vocabulary/commit/56e27069ad50f6a544d51b25df255c954687e51e) for fixing the remaining few broken components. I have added a checklist :ballot_box_with_check: to this [pull request description :information_source:](https://github.com/creativecommons/vocabulary/pull/933) listing all of the components so we can get an idea of which ones remain to be fixed.

@sadn1ck, to get started, you can:

1. [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) this repository
2. clone it locally
    - `git clone git@github.com:sadn1ck/vocabulary.git`
3. create a new branch (e.g. `my-branch`) based on the `restructure` branch
    - `git checkout -b my-branch restructure` 
4. commit your changes to the new branch
    - `git commit -am "An excellent commit message :sunglasses:"`
5. push your changes to GitHub
    - `git push` 
6. [create a pull request from your fork](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) against the `restructure` branch in this project

If any of the above steps present problems, I am available to assist.

**Comment by brylie** [comment]
@sadn1ck, also feel free to contact me on the [Creative Commons Community Slack](https://slack-signup.creativecommons.org/). If you'd like, you are also welcome to attend this month's [Creative Commons Open Source Community Call](https://codebuddies.org/hangout/3fFKNPEjr2EdyERGz) to discuss this and other projects.

**Comment by sadn1ck** [comment]
> @sadn1ck, also feel free to contact me on the [Creative Commons Community Slack](https://slack-signup.creativecommons.org/). If you'd like, you are also welcome to attend this month's [Creative Commons Open Source Community Call](https://codebuddies.org/hangout/3fFKNPEjr2EdyERGz) to discuss this and other projects.

Yes I have joined the Slack, will contact you there if I have any issues!

**Comment by brylie** [comment]
> Yes I have joined the Slack, will contact you there if I have any issues!

Very cool! By the way, I see that you have made a considerable amount of contributions to projects on GitHub. Please excuse me for making the "getting started" checklist above, if all of those steps are familiar to you. :smiley:

**Comment by sadn1ck** [comment]
> > Yes I have joined the Slack, will contact you there if I have any issues!
> 
> Very cool! By the way, I see that you have made a considerable amount of contributions to projects on GitHub. Please excuse me for making the "getting started" checklist above, if all of those steps are familiar to you. 😃

haha all good! :D 

I have made a WIP PR https://github.com/creativecommons/vocabulary/pull/940 here, can you do a small review when you have time? Thanks!

File: scripts/build_fonts.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Run build
+vue-cli-service build --target lib --name fonts --dest ./dist/fonts --entry ./src/fonts.js
+
+echo -n "Copying meta files..."
+cp README.md dist/fonts
+cp LICENSE dist/fonts
+cp packages/fonts.json dist/fonts/package.json
```
Comment (brylie) [inline scripts/build_fonts.sh:9]: This causes an error:

> Copying meta files...cp: cannot stat 'packages/fonts.json': No such file or directory

File: scripts/build_vocabulary.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Run build
+vue-cli-service build --target lib --name vocabulary --dest ./dist/vocabulary --entry ./src/vocabulary.js
+
+echo -n "Copying meta files..."
+cp README.md dist/vocabulary
+cp LICENSE dist/vocabulary
+cp packages/vocabulary.json dist/vocabulary/package.json
```
Comment (brylie) [inline scripts/build_vocabulary.sh:9]: Error here:

> Copying meta files...cp: cannot stat 'packages/vocabulary.json': No such file or directory

File: scripts/build_vue_vocabulary.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Run build
+vue-cli-service build --target lib --name vue-vocabulary --dest ./dist/vue-vocabulary --entry ./src/vue_vocabulary.js
+
+echo -n "Copying meta files..."
+cp README.md dist/vue-vocabulary
+cp LICENSE dist/vue-vocabulary
+cp packages/vue-vocabulary.json dist/vue-vocabulary/package.json
```
Comment (brylie) [inline scripts/build_vue_vocabulary.sh:9]: Error here:

> Copying meta files...cp: cannot stat 'packages/vue-vocabulary.json': No such file or directory

**Review by brylie — CHANGES_REQUESTED** [review]
We should probably locate the files `fonts.json`, `vocabulary.json`, and `vue-vocabulary.json`, since they are needed for their respective `package.json` during the build process.

**Comment by brylie** [comment]
@dhruvkb this pull request is nearly finished :smiley: I've just made a few small changes, so please remember to sync your local branch.

Would you please help resolve the few changes requested?

**Comment by sadn1ck** [comment]
I don't think we need the custom build scripts now since everything is combined, we can use the `build:storybook` command in package.json to output the built files into a directory (`-o`, `docs/` as of now) and statically serve that. You can try that locally, it works for me right now.

```bash
# install deps and stuff

$ npm run build:storybook
$ cd docs
$ npx serve .

# go to displayed localhost URL
```


---

Also, a small issue which I somehow missed

<details>

<summary>Due to the parser text output and some of our custom css having the same class `.value`, some parts of the code have become way too large haha</summary>

image


</details>

**Comment by brylie** [comment]
> I don't think we need the custom build scripts now since everything is combined

We still need to build and publish the packages independently.

**Comment by brylie** [comment]
What would be a good solution for the text size issue?

**Comment by sadn1ck** [comment]
> We still need to build and publish the packages independently.

Ah yes, my bad

> What would be a good solution for the text size issue?

Probably need to update the .value class written by us to something like .custom-value and change wherever it is used 🤔 The styles of `value` class in the code of a component is generated by storybook, I think.

Let's wait for Dhruv's take 😄

**Comment by brylie** [comment]
I'm in favor of deferring the font-size fix until after this pull request is merged. This is already a huge undertaking and we are converging on work in related projects that depend on vocabulary.

@dhruvkb, could you review and approve this pull request if there are no major concerns remaining?

**Comment by dhruvkb** [comment]
I agree with @brylie on fixing the font-size separately. Since it's my own PR I cannot approve it but I've un-drafted it to let it be merged.

**Review by brylie — APPROVED** [review]

**Pull request merged**